source/ProcessManager.Orchestrations/Processes/BRS_021/ForwardMeteredData/ElectricityMarket/MeteringPointMasterDataProvider.cs

```diff
@@ -204,30 +249,72 @@ internal async Task<IReadOnlyCollection<PMMeteringPointMasterData>> GetMasterDat
-                    MeteringPointMasterDataMapper.ConnectionStateMap.Map(meteringPointMasterData.ConnectionState),
-                    MeteringPointMasterDataMapper.MeteringPointTypeMap.Map(meteringPointMasterData.Type),
-                    MeteringPointMasterDataMapper.MeteringPointSubTypeMap.Map(meteringPointMasterData.SubType),
-                    MeteringPointMasterDataMapper.ResolutionMap.Map(meteringPointMasterData.Resolution.Value),
-                    MeteringPointMasterDataMapper.MeasureUnitMap.Map(meteringPointMasterData.Unit),
-                    meteringPointMasterData.ProductId.ToString(),
-                    meteringPointMasterData.ParentIdentification is null
-                        ? null
-                        : new MeteringPointId(meteringPointMasterData.ParentIdentification.Value),
-                    ActorNumber.Create(meteringPointEnergySupplier.EnergySupplier)))
-            .ToList()
-            .AsReadOnly(),
+    private (IReadOnlyCollection<PMMeteringPointMasterData> MeteringPointMasterData,
+        IReadOnlyCollection<MeteringPointMasterDataInconsistencyException> Exceptions) FlattenMasterDataForParent(
+            MeteringPointMasterData meteringPointMasterData) =>
+        (meteringPointMasterData.EnergySuppliers
+                .Select(
+                    meteringPointEnergySupplier =>
+                        new PMMeteringPointMasterData(
+                            new MeteringPointId(meteringPointMasterData.Identification.Value),
+                            meteringPointEnergySupplier.StartDate.ToDateTimeOffset(),
+                            meteringPointEnergySupplier.EndDate.ToDateTimeOffset(),
+                            new PMGridAreaCode(meteringPointMasterData.GridAreaCode.Value),
+                            ActorNumber.Create(meteringPointMasterData.GridAccessProvider),
+                            meteringPointMasterData.NeighborGridAreaOwners,
+                            MeteringPointMasterDataMapper.ConnectionStateMap.Map(
+                                meteringPointMasterData.ConnectionState),
+                            MeteringPointMasterDataMapper.MeteringPointTypeMap.Map(
+                                meteringPointMasterData.Type),
+                            MeteringPointMasterDataMapper.MeteringPointSubTypeMap.Map(
+                                meteringPointMasterData.SubType),
+                            MeteringPointMasterDataMapper.ResolutionMap.Map(meteringPointMasterData.Resolution.Value),
+                            MeteringPointMasterDataMapper.MeasureUnitMap.Map(meteringPointMasterData.Unit),
+                            meteringPointMasterData.ProductId.ToString(),
+                            meteringPointMasterData.ParentIdentification is null
+                                ? null
+                                : new MeteringPointId(meteringPointMasterData.ParentIdentification.Value),
```

Comment:
```suggestion
                            null
```
As of right now, the only way to call "FlattenMasterDataForParent" is if "meteringPointMasterData.ParentIdentification" is null?